src/storage/v3/shard_rsm.cpp

```diff
@@ -86,56 +85,55 @@ std::optional<std::map<PropertyId, Value>> CollectSpecificPropertiesFromAccessor
       return std::nullopt;
     }
     auto &value = result.GetValue();
-    if (value.IsNull()) {
-      spdlog::debug("The specified property does not exist but it should");
-      return std::nullopt;
-    }
-    ret.emplace(std::make_pair(prop, FromPropertyValueToValue(value)));
+    ret.emplace(std::make_pair(prop, FromPropertyValueToValue(std::move(value))));
```

Comment:
Just for future reference: returning `Null` is totally fine, because not all vertex will have all the requested properties, but that is fine.